### PR TITLE
Fixes for --no-restore still doing a restore

### LINKED ISSUE #3485: Paket restore runs on dotnet build --norestore
### Description

When calling dotnet build with the --norestore option a paket restore is still run.

### Repro steps

On any repo, call donet build --norestore.

### Expected behavior

Paket is not called

### Actual behavior

Paket is called

### Known workarounds

None

### PR DESCRIPTION
**Pull request by BlythMeister**
Fixes #3485 and #3449

The version number used in the PaketPropsVersion in https://github.com/fsprojects/Paket/blob/master/src/Paket.Core/Installation/RestoreProcess.fs#L328 is `5.185.3` which does not match the targets.
This means that `PaketRestoreRequired` is always true, even if you have literally just built.

### CODE REVIEW

**Comment by baronfel** [comment]
Is there a way to template this value so that it could be populated on release builds? I'm just worried about reintroducing this failure transparently down the line.

Either that or a more low tech fix: a test that checks if the version in the embedded targets matches the current assembly version?

**Comment by BlythMeister** [comment]
@baronfel have added comments above the 2 values which need to be in sync.

These appear to be values for the format of the data since the value in the props does not match that of paket itself.

Have also made an additional change to the `beforetarget` to make paket mirror the order of the standard nuget restore target.

**Comment by BlythMeister** [comment]
Scrap that before target stuff...that's just plain wrong...will rebase and fix up this PR :D

File: src/Paket.Core/embedded/Paket.Restore.targets [hunk 1]
```diff
@@ -102,7 +102,11 @@
       <PaketRestoreRequired Condition=" '$(PaketRestoreLockFileHash)' == '' ">true</PaketRestoreRequired>
     </PropertyGroup>
 
-    <PropertyGroup Condition="'$(PaketPropsVersion)' != '5.174.2' ">
+	<!-- 
+		This value should match the version in the props generated by paket 
+		If they differ, this means we need to do a restore in order to ensure correct dependencies
+	-->
+    <PropertyGroup Condition="'$(PaketPropsVersion)' != '5.185.3' ">
```
Comment (forki) [inline src/Paket.Core/embedded/Paket.Restore.targets]: I don't think this is correct. This is the props file format version  - and it did not change recently.

Comment (forki) [reply src/Paket.Core/embedded/Paket.Restore.targets]: maybe we need to check if it's not older than the version in the targets file. dunno

Comment (BlythMeister) [reply src/Paket.Core/embedded/Paket.Restore.targets]: this commit is when the props version was changed. https://github.com/fsprojects/Paket/commit/828f9e4553d4d242be603e72a08cdec0c5fb454f
This file was not updated at that time.

**Comment by forki** [comment]
Ouch. Good catch

Am Fr., 18. Jan. 2019, 10:16 hat Chris Blyth <notifications@github.com>
geschrieben:

**Comment by BlythMeister** [comment]
i'm an experienced archaeology driven developer ;)

**Pull request merged**